dspace_metadata_import.sh

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env bash
+
+
+########################################################################################
+## Run this script to import metadata from old dspace into new dspace6 database, and  ##
+## copy over the metadata registry files.											  ##
+########################################################################################
+
+
+## Refer to : https://unix.stackexchange.com/questions/191990/shell-script-to-execute-psql-command ##
+## DSpace6 Upgrading: https://wiki.duraspace.org/display/DSDOC6x/Upgrading+DSpace ##
+
+
+#########################################################################
+#                       									 			#
+#	Need to 															#
+#		1) copy over the dumped pgsql file from old dspace, and			#
+#		2) set up the ownership, and									#
+#		3) replace the user to be dspace or something, and 				#
+#		4) comment out the block of adding rule protect_approval_upd	#
+#    	5) comment out the privileges granted to user libacct			#
+#                       												#
+#########################################################################
+
+PATH_TO_REGISTRIES_FROM_OLD_DSPACE=/tmp/registries.tar.gz
+PATH_TO_OLD_DSPACE_DATABASE_DATA=/tmp/test3.pgsql
+EXISTING_DSPACE_DATABASE_NAME=dspace
+EXISTING_DSPACE_DATABASE_NAME_CHANGED=dspace_old
+NEW_DSPACE_DATABASE_NAME=dspace2
+DSPACE_DATABASE_OWNER=dspace
+
+sudo systemctl stop tomcat
+
+echo "#####  Create new database with a new name  #####"
+
+sudo -u postgres psql <<EOF
+
+DROP DATABASE IF EXISTS $NEW_DSPACE_DATABASE_NAME;
+
+DROP DATABASE IF EXISTS $EXISTING_DSPACE_DATABASE_NAME_CHANGED;
+
+CREATE DATABASE $NEW_DSPACE_DATABASE_NAME WITH OWNER $DSPACE_DATABASE_OWNER; 
+
+\c $NEW_DSPACE_DATABASE_NAME
+
+CREATE EXTENSION pgcrypto;
+
+EOF
+
+echo "#####  Import metadata into the new database  #####"
+
+sudo -u postgres psql -U $DSPACE_DATABASE_OWNER -h localhost $NEW_DSPACE_DATABASE_NAME < $PATH_TO_OLD_DSPACE_DATABASE_DATA
```

Comment:
We'll need to make sure that we don't hard code for "localhost". Both of our target systems need to be able to run against RDS.  Thinking about it more, that means you won't be able to rely on ident-based authentication, we'll need to do password based auth.

Reply:
Not sure how to do password based auth